tasks/test_helper/builders/data_handling/data_pre_process.py

```diff
@@ -0,0 +1,58 @@
+import random as rand
+
+import h5py
+import numpy as np
+import scipy.sparse
+from scipy.sparse import csc_array, random
+
+
+def simulate_csc_h5_data(
+        file_path: str = None,
+        filename: str = 'simulated_csc_h5_data.h5'
+        ) -> (scipy.sparse, tuple[int, int]):
+    """
+    ### Function that simulates h5-file and corresponding matrix for comparison and testing of functions.
+    ---
+    #### Args:
+        - file_path (str): Path to folder where h5-file will be stored. Default None
+        - filename (str): Name of h5-file. Default 'simulated_csc_h5_data.h5'
+    ---
+    #### Returns:
+        - Transpose of sparse matrix
+    """
+
+    shape = (rand.randint(50, 1000), rand.randint(50, 1000))
+    simulated_matrix = random(m=shape[0],
+                              n=shape[1],
+                              density=0.1)
+
+    sparse_representation = csc_array(simulated_matrix)
+    simulated_indices = sparse_representation.indices
+    simulated_indptr = sparse_representation.indptr
+    simulated_data = sparse_representation.data
+    amount_mt = int(0.05*shape[0])
+    gene_names = ['g'] * shape[0]
+    cell_names = [f'C{i}' for i in range(1, shape[1] + 1)]
+    id_number = [f'Id{i}' for i in range(1, shape[0]+1)]
+    feature_type = ['Gene Expression'] * shape[0]
+    genome = ['GRCh38'] * shape[0]
+    random_indices = rand.sample(range(shape[0]), amount_mt)
+
+    # Change the values at the selected indices to 'MT'
+    for index in random_indices:
```

Comment:
Avoid using loops as much as possible, they tend to get inefficient quickly. Instead, use built-in options (see example)
Code:
`a = np.array(['g','g','g','g','g','g'],dtype=object)

a[[1,3,5]] = "ta"`

![image](https://github.com/lozadaa-chalmers/transformer-bio/assets/144219103/e86d7cf1-444c-42cf-8a31-b06683256f9b)

Reply:
Thanks! Will do that.